OpenRA.Mods.Common/Traits/Modifiers/FrozenUnderFog.cs

```diff
@@ -70,18 +70,24 @@ void INotifyCreated.Created(Actor self)
 			{
 				var frozenActor = new FrozenActor(self, this, footprint, player, startsRevealed);
 				player.PlayerActor.Trait<FrozenActorLayer>().Add(frozenActor);
-				return new FrozenState(frozenActor) { IsVisible = startsRevealed };
+				return new FrozenState(frozenActor) { IsVisible = !frozenActor.Visible };
 			});
 
 			// Set the initial visibility state
 			// This relies on actor.GetTargetablePositions(), which is also setup up in Created.
 			// Since we can't be sure whether our method will run after theirs, defer by a frame.
-			if (startsRevealed)
-				self.World.AddFrameEndTask(_ =>
+			self.World.AddFrameEndTask(_ =>
+			{
+				for (var playerIndex = 0; playerIndex < frozenStates.Count; playerIndex++)
 				{
-					for (var playerIndex = 0; playerIndex < frozenStates.Count; playerIndex++)
-						UpdateFrozenActor(frozenStates[playerIndex].FrozenActor, playerIndex);
-				});
+					var state = frozenStates[playerIndex];
+					if (startsRevealed || state.IsVisible)
+					{
+						UpdateFrozenActor(state.FrozenActor, playerIndex);
+						state.FrozenActor.Hidden = false;
```

Comment:
Why do we have to set this manually? Shouldn't this be covered by https://github.com/OpenRA/OpenRA/blob/aa14c9c570a8ac3856dceca94025b7a48bce0a95/OpenRA.Game/Traits/Player/FrozenActorLayer.cs#L127?

Reply:
That line will set Hidden to `true` during the setup. So we need to manually change it to false.

This is required for tooltips to be visible on the frozen actor. If you comment out that line you will find tooltips no longer appear for the frozen actor.

Reply:
Ah I see so this is for `startsRevealed`. (Because if `state.IsVisible` is `true` `Hidden` shouldn't be set to `true` right?)
Another comment here might be good, or we change it to `state.FrozenActor.Hidden = !startsRevealed;` which should also make things clearer.

Reply:
Added a comment to it.

I found the following behaviour works with `state.FrozenActor.Hidden = false;` but breaks with `state.FrozenActor.Hidden = !startsRevealed;`

- A building is created in line of sight
- You leave line of sight
- Hovering the frozen building produces a tooltip.

The tooltip will be missing with the suggested change.

I too don't find it intuitive, so at this point I'm just aiming for getting the testcases to all pass.

Reply:
Hm ok, that actually sounds like something else is bogus here. We don't need to fix it here Imho.